### ISSUE TITLE
MessageBodyReader not found

### ISSUE THREAD

**Issue by Reillusion** [WojciechZankowski/iextrading4j#22]
## Summary

Hello,

When trying to execute sample code from [wiki](https://github.com/WojciechZankowski/iextrading4j/wiki/Chart) I've encountered an error.

### Error
```
MessageBodyReader not found for media type=application/json;charset=utf-8, type=interface
java.util.List, genericType=java.util.List<pl.zankowski.iextrading4j.api.stocks.Chart>
```
### Code
```
final IEXTradingClient iexTradingClient = IEXTradingClient.create();
final List<Chart> chartList = iexTradingClient.executeRequest(new ChartRequestBuilder()
		.withChartRange(ChartRange.ONE_MONTH)
		.withSymbol("AAPL")
		.build());
```
### Version

```
<dependency>
            <groupId>pl.zankowski</groupId>
            <artifactId>iextrading4j-all</artifactId>
            <version>2.1.8</version>
</dependency>
```

**Comment by WojciechZankowski** [comment]
Hello!

Does it happen on every try? Build is passing so it should be working. I will check in details soon!

Thank you,

Wojtek

**Comment by Reillusion** [comment]
Hello Wojtek,

Unfortunately yes. I will try to recreate this issue on a fresh project with cleared maven cache - possibly problem is in configuration. I will get back to you in a matter of hours with more data. 

Best regards,
Zyskowski, W.

**Comment by WojciechZankowski** [comment]
If you could paste here whole error maybe I would be able to see what is the real reason behind the issue. 

Thanks!

**Comment by Reillusion** [comment]
Yes, obviously. Sorry, should have attached that earlier.

```
MessageBodyReader not found for media type=application/json;charset=utf-8, type=interface java.util.List, genericType=java.util.List<pl.zankowski.iextrading4j.api.stocks.Chart>.
java.lang.IllegalStateException
	at pl.zankowski.iextrading4j.client.rest.manager.RestManager.executeRequest(RestManager.java:41)
	at pl.zankowski.iextrading4j.client.rest.AbstractRestEndpoint.execute(AbstractRestEndpoint.java:16)
	at pl.zankowski.iextrading4j.client.rest.endpoint.GenericRestEndpoint.executeRequest(GenericRestEndpoint.java:14)
	at pl.zankowski.iextrading4j.client.IEXTradingClient.executeRequest(IEXTradingClient.java:35)
	at tests.stock.history.HistoricalPriceServiceTest.IEXTradingTest(HistoricalPriceServiceTest.java:29)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:53)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:123)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:104)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:164)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:110)
	at org.apache.maven.surefire.booter.SurefireStarter.invokeProvider(SurefireStarter.java:175)
	at org.apache.maven.surefire.booter.SurefireStarter.runSuitesInProcessWhenForked(SurefireStarter.java:107)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:68)
Caused by: org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException: MessageBodyReader not found for media type=application/json;charset=utf-8, type=interface java.util.List, genericType=java.util.List<pl.zankowski.iextrading4j.api.stocks.Chart>.
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:231)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:155)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1085)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:874)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:834)
	at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:368)
	at org.glassfish.jersey.client.InboundJaxrsResponse$2.call(InboundJaxrsResponse.java:126)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:419)
	at org.glassfish.jersey.client.InboundJaxrsResponse.runInScopeIfPossible(InboundJaxrsResponse.java:267)
	at org.glassfish.jersey.client.InboundJaxrsResponse.readEntity(InboundJaxrsResponse.java:123)
	at pl.zankowski.iextrading4j.client.rest.manager.RestManager.executeRequest(RestManager.java:38)
	... 33 more
```

**Comment by Reillusion** [comment]
Hello,

Execution on fresh project does not fail. Issue occurred only after I've added following dependency. 

```
<dependency>
      <groupId>com.jimmoores</groupId>
      <artifactId>quandl-core</artifactId>
      <version>2.0.0</version>
</dependency>
```

After removing it from original project previously attached piece of code executes without error but downloaded list is populated by Chart objects with only null fields, no data is retrieved.

**Comment by WojciechZankowski** [comment]
Hey,

Good to know that other library was the cause of the problem.

I see that toString() method in Chart object is done wrong, so thats why you dont see all the data. Not every field is added to toString() method. Try to display data manually and you will see that data is there.

```java
for (final Chart chart : chartList) {
            System.out.println(chart.getDate());
            System.out.println(chart.getOpen());
            System.out.println(chart.getHigh());
            System.out.println(chart.getLow());
            System.out.println(chart.getVolume());
}
```

Basically you will have data in fields that are listed here https://api.iextrading.com/1.0/stock/AAPL/chart/1m

I will keep this issue open to improve toString() method


Thanks,

Wojtek

**Comment by Reillusion** [comment]
Hey,

Thanks! That was way more trivial than I've expected. 

Best regards,
Zyskowski, W.

**Comment by WojciechZankowski** [comment]
I fixed toString method in Chart object and improved overall exception handling in API calls. 

It is available in version 2.2.0 with some other features. 

https://github.com/WojciechZankowski/iextrading4j/releases/tag/IT4J_RELEASE_2_2_0

```java
<dependency>
            <groupId>pl.zankowski</groupId>
            <artifactId>iextrading4j-all</artifactId>
            <version>2.2.0</version>
</dependency>
```

I close the issue. Thanks for your input!
